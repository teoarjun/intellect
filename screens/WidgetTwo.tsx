import * as React from 'react';
import {
  StyleSheet,
  View,
  Text,
  Button,
  TouchableOpacity,
  TouchableHighlight,
} from 'react-native';
import QuestionScreenStyle from '../assets/styles/QuestionScreen';
import * as Colors from '../assets/styles/Colors';
import Icon from 'react-native-vector-icons/Ionicons';

const WidgetTwo = (props: any) => {
  const [answer, setAnswer] = React.useState(1);
  /* istanbul ignore next */
  const bg5 = answer > 4 ? Colors.WHITE : Colors.LIGHT_BLACK;
  /* istanbul ignore next */
  const bg4 = answer > 3 ? Colors.WHITE : Colors.LIGHT_BLACK;
  /* istanbul ignore next */
  const bg3 = answer > 2 ? Colors.WHITE : Colors.LIGHT_BLACK;
  /* istanbul ignore next */
  const bg2 = answer > 1 ? Colors.WHITE : Colors.LIGHT_BLACK;
  /* istanbul ignore next */
  const bg1 = answer > 0 ? Colors.WHITE : Colors.LIGHT_BLACK;
  const moveToScreen = () => {
    /* istanbul ignore next */
    props.navigation.navigate('Home');
  };
  const moveToTwo = () => {
    /* istanbul ignore next */
    props.navigation.navigate('WidgetOne');
  };

  const buttons = () => {
    return (
      <View style={styles.nextButton}>
        <Button title="Next" color="#000000" onPress={moveToTwo} />
      </View>
    );
  };

  const renderLevels = () => {
    return (
      <View style={styles.circleContainer}>
        <TouchableOpacity
          style={[styles.top5, {backgroundColor: bg5}]}
          onPress={() => {
            /* istanbul ignore next */
            setAnswer(5);
          }}
        />
        <TouchableOpacity
          style={[styles.top4, {backgroundColor: bg4}]}
          onPress={() => {
            /* istanbul ignore next */
            setAnswer(4);
          }}
        />
        <TouchableOpacity
          style={[styles.top3, {backgroundColor: bg3}]}
          onPress={() => {
            /* istanbul ignore next */
            setAnswer(3);
          }}
        />
        <TouchableOpacity
          style={[styles.top2, {backgroundColor: bg2}]}
          onPress={() => {
            /* istanbul ignore next */
            setAnswer(2);
          }}
        />
        <TouchableOpacity
          style={[styles.top1, {backgroundColor: bg1}]}
          onPress={() => {
            /* istanbul ignore next */
            setAnswer(1);
          }}
        />
      </View>
    );
  };

  const renderProgressBlock = () => {
    return (
      <View style={styles.progressRow}>
        <View style={styles.activeColor} />
        <View style={styles.inActiveColor} />
        <View style={styles.inActiveColor} />
        <View style={styles.inActiveColor} />
        <View style={styles.inActiveColor} />
      </View>
    );
  };
  const closeButton = () => {
    return <TouchableHighlight style={styles.center} onPress={moveToScreen}>
      <Icon name="close" size={30} />
    </TouchableHighlight>
  }

  const renderQuestionRow = () => {
    return (
      <View style={styles.questionNoRow}>
        <View style={styles.center}>
          <Text style={styles.questionText}>
            Rescue Session: Anger and Frustration
          </Text>
        </View>
        {closeButton()}
      </View>
    );
  };

  const renderQuestion = () => {
    return (
      <View style={styles.titleRow}>
        <Text style={styles.title}>
          Pick your level of anger and frustration?
        </Text>
      </View>
    );
  };

  return (
    <View style={styles.container}>
      {renderProgressBlock()}

      {renderQuestionRow()}
      {renderQuestion()}

      {renderLevels()}

      {buttons()}
    </View>
  );
};

const styles = StyleSheet.create(QuestionScreenStyle);

export default WidgetTwo;
