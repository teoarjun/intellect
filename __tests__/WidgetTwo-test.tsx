// Mock out all top level functions, such as get, put, delete and post:
import React from 'react';
import renderer from 'react-test-renderer';
import WidgetTwo from '../screens/WidgetTwo';

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon')


it('renders correctly with defaults', () => {
  const navigation = {navigate: jest.fn()};
  const button = renderer
    .create(<WidgetTwo name="World" navigation={navigation} />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
