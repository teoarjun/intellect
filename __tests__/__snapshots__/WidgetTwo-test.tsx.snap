// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#96ccd5",
      "flex": 1,
      "padding": 20,
      "paddingTop": 50,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#96ccd5",
        "borderRadius": 20,
        "flexDirection": "row",
        "height": 7,
        "marginVertical": 10,
        "maxWidth": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "width": "18%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#5DABBA",
          "borderRadius": 20,
          "marginHorizontal": 5,
          "width": "18%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#5DABBA",
          "borderRadius": 20,
          "marginHorizontal": 5,
          "width": "18%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#5DABBA",
          "borderRadius": 20,
          "marginHorizontal": 5,
          "width": "18%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#5DABBA",
          "borderRadius": 20,
          "marginHorizontal": 5,
          "width": "18%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 12,
            "textTransform": "uppercase",
          }
        }
      >
        Rescue Session: Anger and Frustration
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <Icon
        name="close"
        size={30}
      />
    </View>
  </View>
  <View
    style={Object {}}
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
        }
      }
    >
      Pick your level of anger and frustration?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginVertical": 20,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.08)",
          "borderBottomLeftRadius": 100,
          "borderBottomRightRadius": 100,
          "borderTopEndRadius": 50,
          "borderTopLeftRadius": 50,
          "height": 40,
          "opacity": 1,
          "width": 250,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.08)",
          "borderBottomLeftRadius": 100,
          "borderBottomRightRadius": 100,
          "borderTopEndRadius": 50,
          "borderTopLeftRadius": 50,
          "height": 40,
          "marginTop": 20,
          "opacity": 1,
          "width": 200,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.08)",
          "borderBottomLeftRadius": 100,
          "borderBottomRightRadius": 100,
          "borderTopEndRadius": 50,
          "borderTopLeftRadius": 50,
          "height": 40,
          "marginTop": 20,
          "opacity": 1,
          "width": 150,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.08)",
          "borderBottomLeftRadius": 100,
          "borderBottomRightRadius": 100,
          "borderTopEndRadius": 50,
          "borderTopLeftRadius": 50,
          "height": 40,
          "marginTop": 20,
          "opacity": 1,
          "width": 100,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomEndRadius": 100,
          "borderBottomLeftRadius": 100,
          "borderBottomRightRadius": 100,
          "borderBottomStartRadius": 100,
          "borderTopEndRadius": 50,
          "borderTopLeftRadius": 50,
          "height": 40,
          "marginTop": 20,
          "opacity": 1,
          "width": 50,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 30,
        "bottom": 20,
        "left": 20,
        "marginVertical": 20,
        "position": "absolute",
        "right": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#000000",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</View>
`;
